File: SPLecture/SPLecture/lecture 9/sigmask-1.py
import signal
import time

if __name__ == "__main__":
    signal.pthread_sigmask(signal.SIG_BLOCK, [signal.SIGINT])

    # code that should not be interrupted
    print("running important code")
    time.sleep(10)
    print("finished running important code")

    signal.pthread_sigmask(signal.SIG_SETMASK, [])

    for i in range(20):
        print(f"counter {i}")
        time.sleep(1)

File: SPLecture/SPLecture/lecture 9/sigmask-1.c
#include <signal.h>
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

int main()
{
    sigset_t blockSet;
    sigset_t prevMask;

    // create an empty set
    sigemptyset(&blockSet);
    // add SIGINT to the empty set
    sigaddset(&blockSet, SIGINT);

    int result = sigprocmask(SIG_BLOCK, &blockSet, &prevMask);
    if (result == -1)
    {
        printf("error with blocking\n");
        exit(-1);
    }
    // code that should not be interrupted
    sleep(10);

    result = sigprocmask(SIG_SETMASK, &prevMask, NULL);
    if (result == -1)
    {
        printf("error with restoring\n");
        exit(-1);
    }

    for (int i = 0; i < 20; i++)
    {
        printf("counter: %d\n", i);
        sleep(1);
    }
}